sympy/combinatorics/rewritingsystem.py

```diff
@@ -135,11 +163,8 @@ def _remove_redundancies(self, changes=False):
             if v != r:
                 del self.rules[r]
                 removed.add(r)
-                if v != w:
-                    new = self.add_rule(v, w)
-                    added.update(new)
-            else:
-                self.rules[v] = w
+            new = self.add_rule(v, w)
```

Comment:
What is the purpose of this line? The rule `v -> w` is in `self.rules` and this call may reintroduce other redundant rules that have already been deleted in the loop.